packages/app/prisma-utils/src/prismaTransaction.ts

```diff
@@ -58,9 +60,12 @@ export const prismaTransaction = (async <T, P extends PrismaClient>(
 		}
 
 		result = await executeTransactionTry(prisma, arg, options)
-		if (result.result || !isRetryAllowed(result, optionsWithDefaults.DbDriver)) {
-			break
-		}
+		if (result.result) break
+
+		const retryAllowed = isRetryAllowed(result, optionsWithDefaults.DbDriver)
+		if (!retryAllowed) break
+
+		if (retryAllowed === 'increase-timeout') optionsWithDefaults.timeout *= 2 // TODO: implement smart timeout increase
```

Comment:
this sounds reasonable, but maybe we should cap it to some max value?

Reply:
Sounds great! we can add a new parameter to define that max value with a default of 30s for example.

About how to increase the timeout, the `*=2` approach sounds good?